ftl/php/builder.py

```diff
@@ -93,8 +105,9 @@ def _gen_composer_install_tar(self, pkg_descriptor, destination_path):
             os.makedirs(app_dir)
 
             # Copy out the relevant package descriptors to a tempdir.
-            ftl_util.descriptor_copy(self._ctx, self._descriptor_files,
-                                     app_dir)
+            if pkg_descriptor is None:
```

Comment:
not sure I understand this part. Why do we only copy if it's none?

Reply:
if there is no pkg_descriptor, that means that composer is being run without a pkg in mind but with a composer.json file in its entirety.  When using the composer.json file, we need to have it copied.  When using a pkg_descriptor, we don't need to copy but can just use the string arg:
composer require <pkg> <version>
VS
composer install   # this uses composer.json in same dir